### PR TITLE
[docs] Add jaeger sampling docs

### PR DESCRIPTION
**Pull request by bmorelli25**
## Motivation/summary

Adds documentation for Jaeger sampling, including limitations and caveats.

## Related issues

For https://github.com/elastic/apm-server/issues/3487.
PR: https://github.com/elastic/apm-server/pull/3490.

### CODE REVIEW

File: docs/jaeger-support.asciidoc [hunk 1]
```diff
@@ -53,6 +53,27 @@ Based on the endpoint enabled in the previous step, configure the relevant host
 * `apm-server.jaeger.grpc.host` defaults to `localhost:14250`.
 * `apm-server.jaeger.http.host` defaults to `localhost:14268`.
 
+[[jaeger-configure-sampling]]
+*Configure Sampling*
+
+The gRPC endpoint supports probabilistic sampling which can be used to reduce the amount of data that your agents collect and send.
+Probabilistic sampling makes a random sampling decision based on the value of `sampling.default`.
+For example, a value of `.2` means that 20% of traces will be sampled.
+Sampling ratios can also be adjusted, on a per service basis, via the Agent configuration page in the APM app.
```
Comment (simitt) [inline docs/jaeger-support.asciidoc]: The logic changed. Now both, service specific and defaults values for all services are fetched from agent configuration management.

File: docs/jaeger-support.asciidoc [hunk 2]
```diff
@@ -53,6 +53,27 @@ Based on the endpoint enabled in the previous step, configure the relevant host
...
+
+To enable sampling:
+
+* Set `apm-server.grpc.sampling.enabled` to `enabled`.
+* Set `apm-server.grpc.sampling.default` to the desired sampling rate (between `0` and `1.0`).
```
Comment (simitt) [inline docs/jaeger-support.asciidoc]: I removed the config parameters, based on https://github.com/elastic/apm-server/pull/3490#discussion_r393410810. The gRPC sampling endpoint will be enabled as soon as `apm-server.jaeger.grpc.enabled: true`.

File: docs/jaeger-support.asciidoc [hunk 3]
```diff
@@ -133,6 +155,19 @@ Defaults to the standard Jaeger gRPC collector port `14250`.
...
+Enable the sampling rate to be fetched from the gRPC endpoint.
+Only probabilistic sampling is supported.
+No authorization token can be configured for this endpoint.
+Sampling is disabled by default.
+
+[float]
+===== `grpc.sampling.default`
+Set the default sampling rate used when no service specific sampling is configured.
+Valid range is from `0` to `1.0`, where `0` samples at 0% and `1.0` samples at 100%.
+Defaults to `1.0`.
```
Comment (simitt) [inline docs/jaeger-support.asciidoc]: Sorry for the additional work here, as specified above, I removed the config options again.

Comment (bmorelli25) [reply docs/jaeger-support.asciidoc]: No problem! That's what I get for starting early on it 😁

**Comment by bmorelli25** [comment]
Updated with the request changes, thanks @simitt! I still might need to add the `[x-pack]` flag, depending on the outcome of #3490.

File: docs/jaeger-support.asciidoc [hunk 4]
```diff
@@ -60,9 +60,19 @@ The gRPC endpoint supports probabilistic sampling, which can be used to reduce t
 Probabilistic sampling makes a random sampling decision based on the configured sampling value.
 For example, a value of `.2` means that 20% of traces will be sampled.
 
-Sampling is automatically enabled when `apm-server.jaeger.grpc.enabled` is set to `true`.
-The default sampling ratio, as well as per-service sampling rates,
-are are configured via the {kibana-ref}/agent-configuration.html[Agent configuration] page in the APM app.
+By default, when using Jaeger Clients over gRPC, two things happen:
+
+* APM Server automatically enables the sampling endpoint when `apm-server.jaeger.grpc.enabled` is set to `true`.
+* Jaeger clients poll APM Server for the configured sampling rate.
```
Comment (simitt) [inline docs/jaeger-support.asciidoc]: The Jaeger agent polls the sampling rate

File: docs/jaeger-support.asciidoc [hunk 5]
```diff
@@ -53,6 +53,34 @@ Based on the endpoint enabled in the previous step, configure the relevant host
...
 * `apm-server.jaeger.http.host` defaults to `localhost:14268`.
 
+[[jaeger-configure-sampling]]
+*Configure Sampling*
+
+The gRPC endpoint supports probabilistic sampling, which can be used to reduce the amount of data that your agents collect and send.
+Probabilistic sampling makes a random sampling decision based on the configured sampling value.
+For example, a value of `.2` means that 20% of traces will be sampled.
+
+By default, when using Jaeger Clients over gRPC, two things happen:
```
Comment (simitt) [inline docs/jaeger-support.asciidoc]: Jaeger Agents talk to the APM Server via gRPC, Jaeger clients can only talk to the APM Server via thrift over http. Sampling is only supported for the gRPC endpoint.

File: docs/jaeger-support.asciidoc [hunk 6]
```diff
@@ -53,6 +53,34 @@ Based on the endpoint enabled in the previous step, configure the relevant host
...
+*Configure Sampling*
+
+The gRPC endpoint supports probabilistic sampling, which can be used to reduce the amount of data that your agents collect and send.
+Probabilistic sampling makes a random sampling decision based on the configured sampling value.
+For example, a value of `.2` means that 20% of traces will be sampled.
+
+By default, when using Jaeger over gRPC, two things happen:
+
+* APM Server automatically enables the sampling endpoint when `apm-server.jaeger.grpc.enabled` is set to `true`.
+* Jaeger agents poll APM Server for the configured sampling rate.
```
Comment (simitt) [inline docs/jaeger-support.asciidoc]: I just realized we miss a reference to the [Kibana config](https://www.elastic.co/guide/en/apm/server/master/setup-kibana-endpoint.html). The sampling endpoint is enabled by default, but Kibana is not and needs to be configured.  
Looking at the current Kibana config docs I think they should also be slightly changed to reflect the Jaeger use case.

Comment (bmorelli25) [reply docs/jaeger-support.asciidoc]: Great catch! I've updated the docs to add a reference to the Kibana config, and to better differentiate between the two different methods for configuring Jaeger sampling.

I'm hesitant to update the Kibana config docs to reference Jaeger though. To me, "central jaeger sampling" is simply a feature of central config. So we should set the tone in the Jaeger docs that "central sampling" is powered by APM Agent configuration. If we do this, the Kibana config documentation doesn't need to mention the Jaeger use case.

I tried to do this in the recent commit. Take a look and let me know what you think

**Pull request merged**